### PR TITLE
fix: session support detection spec compliance

### PR DESCRIPTION
**Pull request by nbbeeken**
Discovered while working on a help ticket the reduce callback in the topology description
doesn't correctly return null if it's in the middle of the list of servers.

### CODE REVIEW

**Review by durran — APPROVED** [review]
LGTM

File: test/unit/core/common.js [hunk 1]
```diff
@@ -11,30 +11,39 @@ class ReplSetFixture {
...
+        __nodejs_mock_server__: true,
+        setName: 'rs',
+        setVersion: 1,
+        electionId: this.electionIds[0],
+        hosts: this.servers.map(server => server.uri()),
+        arbiters: [this.arbiterServer.uri()]
+      });
+
+      this.defineReplSetStates();
+      // this.configureMessageHandlers();
```
Comment (emadum) [inline test/unit/core/common.js:45]: Should this line be commented out? Seems there are some issues in CI.

Comment (nbbeeken) [reply test/unit/core/common.js]: Whoops good catch, let's see if the commit I just pushed fixes it. I made the change on the 4.0 port already to add if statements to gate these extra setup calls.

**Review by emadum — APPROVED** [review]
LGTM 👍

**Pull request merged**